plugins/externalwrench/src/ExternalWrench.cc

```diff
@@ -0,0 +1,173 @@
+#include "ExternalWrench.hh"
+
+int ExternalWrench::count = 0;
+
+//Initializing wrench command
+ExternalWrench::ExternalWrench()
+{
+    model_has_link = false;
+
+    //srand(boost::lexical_cast<float>(std::time(NULL)));
+    srand(rand()%100);
+    color[0] = static_cast <float> (rand()) / static_cast <float> (RAND_MAX);
+    color[1] = static_cast <float> (rand()) / static_cast <float> (RAND_MAX);
+    color[2] = static_cast <float> (rand()) / static_cast <float> (RAND_MAX);
+    color[3] = static_cast <float> (rand()) / static_cast <float> (RAND_MAX);
+
+    wrenchPtr->link_name = "world";
+    wrenchPtr->force.resize(3,0);
+    wrenchPtr->torque.resize(3,0);
+    wrenchPtr->duration = 0.0;
+
+    count++;
+    tick = yarp::os::Time::now();
```

Comment:
I am afraid that in general this  will not give you the expected duration of the external force, unless `gzserver` is executed with the `YARP_CLOCK=/clock` env variable and with the `gazebo_yarp_clock` plugin. If one of this conditions is not true, `yarp::os::Time::now()` will return you the system time, resulting in a duration of the external force in simulation that may be much shorter than expected (depending on the RTF) and even worse a duration that is not reproducible.

Reply:
The proper solution to this is to add a `const gazebo::common::UpdateInfo& _info` argument to the applyWrenches method, and use it to retrieved the simulation time step. See other onUpdate methods in the repo to get an inspiration https://github.com/robotology/gazebo-yarp-plugins/search?q=onUpdate&unscoped_q=onUpdate .

Reply:
Time handling is updated https://github.com/robotology/gazebo-yarp-plugins/pull/418/commits/64199f6306c41ef36020454d52dfa14c72b24a8a